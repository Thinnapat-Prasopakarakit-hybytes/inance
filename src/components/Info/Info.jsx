import React from "react";
import {
  FaMapMarkerAlt,
  FaPhoneAlt,
  FaEnvelope,
  FaTwitter,
  FaInstagram,
} from "react-icons/fa";
import { TfiFacebook, TfiYoutube } from "react-icons/tfi";
import "./Info.scss";
import { Link } from "react-router-dom";

const Info = () => {
  const contactItems = [
    {
      icon: <FaMapMarkerAlt />,
      text: "Lorem Ipsum is simply dummy text",
      link: "/",
    },
    {
      icon: <FaPhoneAlt />,
      text: "+02 1234567890",
      link: "/",
    },
    {
      icon: <FaEnvelope />,
      text: "demo@gmail.com",
      link: "/",
    },
  ];

  const socialItems = [
    { icon: <TfiFacebook />, link: "/" },
    { icon: <FaTwitter />, link: "/" },
    { icon: <TfiYoutube />, link: "/" },
    { icon: <FaInstagram />, link: "/" },
  ];

  return (
    <section className="info_section">
      <div className="container">
        <h4>Get In Touch</h4>
        <div className="row">
          <div className="col-lg-10 mx-auto">
            <div className="info_items">
              <div className="row">
                {contactItems.map((item, idx) => (
                  <div className="col-md-4" key={idx}>
                    <Link to={item.link}>
                      <div className="item">
                        <div className="img-box">{item.icon}</div>
                        <p>{item.text}</p>
                      </div>
                    </Link>
                  </div>
                ))}
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className="social-box">
        <h4>Follow Us</h4>
        <div className="box">
          {socialItems.map((item, idx) => (
            <Link to={item.link} key={idx}>
              {item.icon}
            </Link>
          ))}
        </div>
      </div>
    </section>
  );
};

export default Info;
