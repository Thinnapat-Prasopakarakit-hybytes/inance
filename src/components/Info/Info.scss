@use "../../styles/partials/variables" as *;
@use "../../styles/partials/mixins" as *;

.info_section {
  padding: 75px 0;
  background-color: #0a0f43;
  text-align: center;
  color: $white;

  h4 {
    text-transform: uppercase;
    font-weight: bold;
    margin-bottom: 25px;
  }

  .social-box {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin-top: 25px;

    .box {
      background-color: $white;
      padding: 5px 15px;
    }
  }

  a {
    margin: 0 5px;
    color: $primary1;

    i {
      font-size: 18px;
    }

    &:hover {
      color: $primary2;
    }
  }
}

.info_items {
  position: relative;

  a {
    position: relative;
  }

  &::before {
    content: "";
    position: absolute;
    top: 65px;
    width: 75%;
    height: 1px;
    background-color: $white;
    left: 50%;
    transform: translateX(-50%);
  }
  @include tablet {
    &::before {
      width: 1px;
      height: 90%;
      top: 25px;
    }
  }

  .item {
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;

    .img-box {
      width: 80px;
      height: 80px;
      border-radius: 100%;
      background-color: $white;
      background-repeat: no-repeat;
      background-position: center;
      text-align: center;
      line-height: 80px;
      background-color: $primary1;
      color: $white;

      svg {
        font-size: 24px;
      }
    }

    p {
      margin-top: 25px;
      color: #fff;
      margin-bottom: 0;
      background-color: #0a0f43;
    }

    &:hover {
      .img-box {
        background-color: $primary2;
      }
    }
  }
}
