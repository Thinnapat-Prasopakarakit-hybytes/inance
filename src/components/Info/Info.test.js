import React from "react";
import { render, screen } from "@testing-library/react";
import Info from "./Info";
import "@testing-library/jest-dom";
import { BrowserRouter } from "react-router-dom";

jest.mock("react-icons/fa", () => ({
  FaMapMarkerAlt: () => <div data-testid="map-marker-icon" />,
  FaPhoneAlt: () => <div data-testid="phone-icon" />,
  FaEnvelope: () => <div data-testid="envelope-icon" />,
  FaTwitter: () => <div data-testid="twitter-icon" />,
  FaInstagram: () => <div data-testid="instagram-icon" />,
}));

jest.mock("react-icons/tfi", () => ({
  TfiFacebook: () => <div data-testid="facebook-icon" />,
  TfiYoutube: () => <div data-testid="youtube-icon" />,
}));

const renderInfo = () => {
  return render(
    <BrowserRouter>
      <Info />
    </BrowserRouter>
  );
};

describe("Info Component", () => {
  it("should render the component", () => {
    renderInfo();
  });

  it("should render the main headings", () => {
    renderInfo();
    expect(screen.getByText("Get In Touch")).toBeInTheDocument();
    expect(screen.getByText("Follow Us")).toBeInTheDocument();
  });

  it("should render contact items with correct icons and text", () => {
    renderInfo();

    const contactItems = screen.getAllByText(
      /Lorem Ipsum is simply dummy text|demo@gmail.com|\+02 1234567890/
    );
    expect(contactItems).toHaveLength(3);

    expect(screen.getByTestId("map-marker-icon")).toBeInTheDocument();
    expect(screen.getByTestId("phone-icon")).toBeInTheDocument();
    expect(screen.getByTestId("envelope-icon")).toBeInTheDocument();

    const contactLinks = screen.getAllByRole("link");
    expect(contactLinks).toHaveLength(7);
  });

  it("should render social icons with correct links", () => {
    renderInfo();

    const socialIcons = [
      screen.getByTestId("facebook-icon"),
      screen.getByTestId("twitter-icon"),
      screen.getByTestId("youtube-icon"),
      screen.getByTestId("instagram-icon"),
    ];
    socialIcons.forEach((icon) => expect(icon).toBeInTheDocument());

    const socialLinks = screen.getAllByRole("link");
    expect(socialLinks).toHaveLength(7);
  });

  it("should match snapshot", () => {
    const { asFragment } = renderInfo();
    expect(asFragment()).toMatchSnapshot();
  });
});
