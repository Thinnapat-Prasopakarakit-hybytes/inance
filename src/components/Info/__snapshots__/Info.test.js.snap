// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Component should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="info_section"
  >
    <div
      class="container"
    >
      <h4>
        Get In Touch
      </h4>
      <div
        class="row"
      >
        <div
          class="col-lg-10 mx-auto"
        >
          <div
            class="info_items"
          >
            <div
              class="row"
            >
              <div
                class="col-md-4"
              >
                <a
                  href="/"
                >
                  <div
                    class="item"
                  >
                    <div
                      class="img-box"
                    >
                      <div
                        data-testid="map-marker-icon"
                      />
                    </div>
                    <p>
                      Lorem Ipsum is simply dummy text
                    </p>
                  </div>
                </a>
              </div>
              <div
                class="col-md-4"
              >
                <a
                  href="/"
                >
                  <div
                    class="item"
                  >
                    <div
                      class="img-box"
                    >
                      <div
                        data-testid="phone-icon"
                      />
                    </div>
                    <p>
                      +02 1234567890
                    </p>
                  </div>
                </a>
              </div>
              <div
                class="col-md-4"
              >
                <a
                  href="/"
                >
                  <div
                    class="item"
                  >
                    <div
                      class="img-box"
                    >
                      <div
                        data-testid="envelope-icon"
                      />
                    </div>
                    <p>
                      demo@gmail.com
                    </p>
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="social-box"
    >
      <h4>
        Follow Us
      </h4>
      <div
        class="box"
      >
        <a
          href="/"
        >
          <div
            data-testid="facebook-icon"
          />
        </a>
        <a
          href="/"
        >
          <div
            data-testid="twitter-icon"
          />
        </a>
        <a
          href="/"
        >
          <div
            data-testid="youtube-icon"
          />
        </a>
        <a
          href="/"
        >
          <div
            data-testid="instagram-icon"
          />
        </a>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
